src/lib/common/JsonHelper.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+*
+* Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms
+* of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+#include "common/JsonHelper.h"
+
+std::string toJsonString(const std::string& input)
+{
+  std::ostringstream ss;
+  ss << '"';
+  for (std::string::const_iterator iter = input.begin(); iter != input.end(); ++iter) {
+        switch (*iter) {
+            case '\\': ss << "\\\\"; break;
+            case '"': ss << "\\\""; break;
+            case '/': ss << "\\/"; break;
+            case '\b': ss << "\\b"; break;
+            case '\f': ss << "\\f"; break;
+            case '\n': ss << "\\n"; break;
+            case '\r': ss << "\\r"; break;
+            case '\t': ss << "\\t"; break;
+            default: ss << *iter; break;
+      }
+  }
+  ss << '"';
+  return ss.str();
+}
```

Comment:
No, it does not mean that at all.
Normally a user would know what is a sufficient amount of data, and if not, just allocate a char buffer of twice the size of the buffer to be replaced. Mas size of the output buffer is 2 * strlen(inBuffer).

It is common praxis for a function of a library to not decide for its user. If the user wants to use a 'slow' heap variable (slow due to the call to malloc), let him do that, if he prefers to use a faster allocation on the stack, he can do that too.
I would strongly recommend to use the stack if possible. What I would pretty much never do in a function of a library is to not let this decision be up to the user.